presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2054,6 +2054,30 @@ public void testJoinWithConstantExpression()
     }
 
     @Test
+    public void testJoinWithConstantExpressionWithCoercion()
+    {
+        // Covers #7520
+        {
+            MaterializedResult actual = computeActual("select count(*) > 0 from nation join region on (cast ('a' as char(1)) = cast ('a' as char(2)))");
```

Comment:
There's a difference between how H2 and Presto handle char(1) = char(2).
See, for:
```java
assertQuery("select (cast ('a' as char(1)) = cast ('a' as char(2)))");
```
we have:
```sql
java.lang.AssertionError: For query: 
 select (cast ('a' as char(1)) = cast ('a' as char(2)))
not equal
Actual 1 rows:
    [false]
Expected 1 rows:
    [true]
```

I'll put a comment about it in the test.